README.md

```diff
@@ -1,10 +1,177 @@
 # bmc-experiments
 
-This repository hosts experimental Rust firmware code for the BMC (and related) microcontrollers.
+This repository hosts experimental Rust firmware code for the BMC (and related)
+microcontrollers. The research conducted here is focused on enabling Racklet BMC
+functionality, such as hosting boot partition files, performing cryptographic
+verification, and logging traces on ARM Cortex-M microcontrollers. We're all
+still learning, and the embedded Rust ecosystem is still quite young, so feel
+free to join the pioneering!
+
+## Usage
+
+The experiments in this repository currently target the [ATSAMD51G19A]
+microcontroller (MCU) (on an [Adafruit ItsyBitsy M4 Express]) as it was readily
+available, and has decent community support and enough resources for learning
+and experimentation. While the final chip used for the BMC boards is still
+undecided, the long term goal for the firmware and libraries here is to be
+somewhat vendor and chip agnostic using abstraction layers and feature gates.
+The core team doesn't have a wide variety of hardware available, but if you can
+put in the effort to support and test the builds on your particular
+microcontroller board, we're more than happy enable support for it!
+
+[ATSAMD51G19A]: https://www.microchip.com/en-us/product/ATSAMD51G19A
+
+[Adafruit ItsyBitsy M4 Express]: https://www.adafruit.com/product/3800
+
+### Running and debugging the experiments
+
+This repository uses [`probe-run`] to enable running (and to some extent
+debugging) embedded applications just like native ones using `cargo run`. This
+includes support for highly performant stdin/stdout and automatic panic message
+output using the [Real-Time Transfer (RTT)] I/O protocol from the [`probe-rs`]
+project. For more serious debugging a configuration for [`cargo-embed`] is
+provided, which will provide a GDB server when run with `cargo embed debug
+<args>` which can then be attached to using a multi-arch GDB build either from
+the CLI or your favorite IDE.
+
+[`probe-rs`]: https://probe.rs/
+
+[`probe-run`]: https://github.com/knurling-rs/probe-run
+
+[`cargo-embed`]: https://github.com/probe-rs/cargo-embed
+
+[Real-Time Transfer (RTT)]: https://github.com/probe-rs/probe-rs-rtt
+
+### Debug Probe needed
```

Comment:
what struck me is that it'd be nice to use draw.io/diagrams.net here for visualizing e.g. IDE <=> gdb <=> openocd <=> debug probe <=> MCU, along with all the protocols supported at each stage.